File: sdkwork-ai-sdk-python/sdkwork_ai_sdk/api/response.py
from typing import Any, Dict, List, Optional
from ..http_client import HttpClient
from ..models import CancelResponseResponse, CreateResponsePostRequest, CreateResponsePostResponse, DeleteResponseDeleteResponse, GetResponseGetResponse, ListResponseInputItemsGetResponse

class ResponseApi:
    """response API client."""
    
    def __init__(self, client: HttpClient):
        self._client = client

    def cancel(self, response_id: str) -> CancelResponseResponse:
        """Cancel response"""
        return self._client.post(f"/ai/v3/responses/{response_id}/cancel")

    def create_response(self, body: CreateResponsePostRequest) -> CreateResponsePostResponse:
        """Create response"""
        return self._client.post(f"/ai/v3/responses", json=body)

    def list_response_input_items(self, response_id: str) -> ListResponseInputItemsGetResponse:
        """List response input items"""
        return self._client.get(f"/ai/v3/responses/{response_id}/input_items")

    def get_response(self, response_id: str) -> GetResponseGetResponse:
        """Get response"""
        return self._client.get(f"/ai/v3/responses/{response_id}")

    def delete_response(self, response_id: str) -> DeleteResponseDeleteResponse:
        """Delete response"""
        return self._client.delete(f"/ai/v3/responses/{response_id}")

File: sdkwork-ai-sdk-flutter/lib/src/api/response.dart
import '../http/client.dart';
import '../models.dart';

class ResponseApi {
  final HttpClient _client;
  
  ResponseApi(this._client);

  /// Cancel response
  Future<CancelResponseResponse?> cancel(String response_id) async {
    final response = await _client.post(ApiPaths.aiPath('/responses/${response_id}/cancel'));
    return response is CancelResponseResponse ? response : null;
  }

  /// Create response
  Future<CreateResponsePostResponse?> createResponse(CreateResponsePostRequest body) async {
    final response = await _client.post(ApiPaths.aiPath('/responses'), body: body, contentType: 'application/json');
    return response is CreateResponsePostResponse ? response : null;
  }

  /// List response input items
  Future<ListResponseInputItemsGetResponse?> listResponseInputItems(String response_id) async {
    final response = await _client.get(ApiPaths.aiPath('/responses/${response_id}/input_items'));
    return response is ListResponseInputItemsGetResponse ? response : null;
  }

  /// Get response
  Future<GetResponseGetResponse?> getResponse(String response_id) async {
    final response = await _client.get(ApiPaths.aiPath('/responses/${response_id}'));
    return response is GetResponseGetResponse ? response : null;
  }

  /// Delete response
  Future<DeleteResponseDeleteResponse?> deleteResponse(String response_id) async {
    final response = await _client.delete(ApiPaths.aiPath('/responses/${response_id}'));
    return response is DeleteResponseDeleteResponse ? response : null;
  }
}
